src/gui/src/SampleEditor/MainSampleWaveDisplay.cpp

```diff
@@ -199,26 +218,41 @@ void MainSampleWaveDisplay::updateDisplay( const QString& filename )
 	update();
 
 }
+void MainSampleWaveDisplay::mouseUpdateDone() {
+	HydrogenApp::get_instance()->getSampleEditor()->returnAllMainWaveDisplayValues();
+	m_bStartSliderIsMoved = false;
+	m_bLoopSliderIsMoved = false;
+	m_bEndSliderIsmoved = false;
+}
+
 
 void MainSampleWaveDisplay::mouseMoveEvent(QMouseEvent *ev)
 {
-	testPosition( ev );
+	if (ev->buttons() && Qt::LeftButton) {
+		testPosition( ev );
+	} else {
+		chooseSlider( ev );
+	}
 	update();
+	mouseUpdateDone();
```

Comment:
How come you mark the state of the SampleEditor unclean once the user enters the MainSampleWaveDisplay using the mouse? This makes the "Apply Changes" not flat and indicates something has changed. Changes via drag should be already covered by the call of `mouseUpdateDone` in `mousePressEvent`

Reply:
> How come you mark the state of the SampleEditor unclean once the user enters the MainSampleWaveDisplay using the mouse? This makes the "Apply Changes" not flat and indicates something has changed. Changes via drag should be already covered by the call of `mouseUpdateDone` in `mousePressEvent`

Hum, actually, it does not mark the editor as changed, only setClean()/setUnclean() does that. I should maybe rename `mouseUpdateDone()` to something like `mouseActionDone()`